app/controllers/concerns/user_eager_loading.rb

```diff
@@ -0,0 +1,38 @@
+module UserEagerLoading
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def load_current_user(with: [], only: nil)
+      if only
+        load_current_user_only_sometimes(with, only.map(&:to_s))
+      else
+        load_current_user_always(with)
+      end
+    end
+
+    private
+
+    def load_current_user_only_sometimes(with, only)
+      class_eval <<-METHOD, __FILE__, __LINE__ + 1
+        def current_user
+          @current_user ||= begin
+            record = super
+            if #{only.inspect}.include?(action_name)
+              record && User.includes(#{with.inspect}).find(record.id)
+            else
+              record
+            end
+          end
+        end
+      METHOD
+    end
+
+    def load_current_user_always(with)
+      class_eval <<-METHOD, __FILE__, __LINE__ + 1
+        def current_user
+          @current_user ||= super && User.includes(#{with.inspect}).find(@current_user.id)
+        end
+      METHOD
+    end
+  end
+end
```

Comment:
One way to tell if the bullet errors raised during tests are legitimate or not is to try to reproduce in development. Our development environment is set to log and alert on any errors: https://github.com/18F/identity-idp/blob/master/config/environments/development.rb#L30-L35

From the bullet logs, it looks like errors are only being raised in the test environment when saving the user in PersonalKeyGenerator, which happens when you get your personal key during the LOA3 flow. I was not able to reproduce any errors in development, which points to a bug in the bullet gem, or maybe something in our test setup. This could be the bug in question:
https://github.com/flyerhzm/bullet/issues/382

While we wait for a fix, we can ignore this error in the test config like this:
```ruby
config.after_initialize do
  Bullet.enable = true
  Bullet.bullet_logger = true
  Bullet.raise = true
  Bullet.add_whitelist(
    type: :n_plus_one_query, class_name: 'User', association: :phone_configuration
  )
end
```
In general, raising an N+1 query error on a `has_one` association where we are not iterating through more than 1 user seems suspect to me.

Reply:
Cool. I'll add that exception.